.github/workflows/build.yml

```diff
@@ -0,0 +1,200 @@
+        run: mkdir -p ./android/app/src/production ; echo "$FILE_ANDROID_GOOGLE_SERVICES_PRODUCTION" | base64 -d > ./android/app/src/production/google-services.json
+      - name: Decode the dev release keystore file
+        # passwords and secrets used for decrypting this keystore file (in the build step further down)
+        # MAPSWIPE_RELEASE_STORE_FILE=mapswipe-dev-release-key.keystore
+        # MAPSWIPE_RELEASE_KEY_ALIAS=mapswipe-dev-key-alias
+        # MAPSWIPE_RELEASE_STORE_PASSWORD=xxx
+        # MAPSWIPE_RELEASE_KEY_PASSWORD=xxx
+        # All vars are taken from actions env vars, prefixed with ORG_GRADLE_PROJECT_
+        # See https://docs.gradle.org/5.6.4/userguide/build_environment.html#sec:project_properties
+        # The keystore file is a binary file, somehow the base64 decoding used for other files does
+        # not work well here, so we use gpg symetric encryption instead as recommended by github:
+        # https://docs.github.com/en/actions/reference/encrypted-secrets#limits-for-secrets
+        env:
+          FILE_DECRYPTION_PASSPHRASE: ${{ secrets.FILE_DECRYPTION_PASSPHRASE }}
+        run: gpg --quiet --batch --yes --decrypt --passphrase="$FILE_DECRYPTION_PASSPHRASE" --output ./android/app/mapswipe-dev-release-key.keystore ./android/app/mapswipe-dev-release-key.keystore.gpg
+      - name: Decode the production release keystore file
+        env:
+          FILE_DECRYPTION_PASSPHRASE: ${{ secrets.FILE_DECRYPTION_PASSPHRASE }}
+        run: gpg --quiet --batch --yes --decrypt --passphrase="$FILE_DECRYPTION_PASSPHRASE" --output ./android/app/mapswipe-production-release-key.keystore ./android/app/mapswipe-production-release-key.keystore.gpg
+      - name: Get the tag value from github
+        # see https://stackoverflow.com/a/58178121/1138710
+        id: vars
+        run: echo ::set-output name=tag::${GITHUB_REF#refs/*/}
+      - name: Save git version so the app can show it to users
+        env:
+          # we need this hack because github default env vars don't include the tag itself
+          GIT_TAG: ${{ steps.vars.outputs.tag }}
+        run: sed -ri "s/COMMIT/$GITHUB_SHA/" debugInfo.json; sed -ri "s/TAG/$GIT_TAG/" debugInfo.json
+      - name: Jetify react files
+        run: yarn jetifier
+      - name: Increase filesystem inotify watch limit
+        run: sudo sysctl fs.inotify.max_user_watches=524288
+      - name: Build dev APK
+        env:
+          ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_KEY_ALIAS: ${{ secrets.ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_KEY_ALIAS }}
+          ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_KEY_PASSWORD: ${{ secrets.ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_KEY_PASSWORD }}
+          ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_STORE_FILE: ${{ secrets.ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_STORE_FILE }}
+          ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_STORE_PASSWORD: ${{ secrets.ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_STORE_PASSWORD }}
+        run: yarn buildDev
+      - name: Build production APK
```

Comment:
Hey,
so just checking here: We will always build the production APK for every commit? But only upload if it is tagged, but not when the release is tagged with beta?

Reply:
yes, this is what we were doing in travis. For now, I focused on getting the same functionality up and running, but I think we should split things up as you suggested elsewhere, it feels a bit silly to build the app over and over each time.